request_model.go

```diff
@@ -164,3 +164,69 @@ func (id *GoDataIdentifier) GetKey(key string) (string, bool) {
 	v, ok := map[string]string(*id)[key]
 	return v, ok
 }
+
+// GoDataCommonStructure represents either a GoDataQuery or ExpandItem in a uniform manner
+// as a Go interface. This allows the writing of functional logic that can work on either type,
+// such as a provider implementation which starts at the GoDataQuery and walks any nested ExpandItem
+// in an identical manner.
+type GoDataCommonStructure interface {
+	GetPath() string
+	GetFilter() *GoDataFilterQuery
+	GetAt() *GoDataFilterQuery
+	GetApply() *GoDataApplyQuery
+	GetExpand() *GoDataExpandQuery
+	GetSelect() *GoDataSelectQuery
+	GetOrderBy() *GoDataOrderByQuery
+	GetTop() *GoDataTopQuery
+	GetSkip() *GoDataSkipQuery
+	GetCount() *GoDataCountQuery
+	GetInlineCount() *GoDataInlineCountQuery
+	GetSearch() *GoDataSearchQuery
+	GetCompute() *GoDataComputeQuery
+	GetFormat() *GoDataFormatQuery
+	AddExpandItem(*ExpandItem)
+}
+
+// GoDataQuery implementation of GoDataCommonStructure interface
+func (o *GoDataQuery) GetPath() string                         { return "" }
+func (o *GoDataQuery) GetFilter() *GoDataFilterQuery           { return o.Filter }
+func (o *GoDataQuery) GetAt() *GoDataFilterQuery               { return o.At }
+func (o *GoDataQuery) GetApply() *GoDataApplyQuery             { return o.Apply }
+func (o *GoDataQuery) GetExpand() *GoDataExpandQuery           { return o.Expand }
+func (o *GoDataQuery) GetSelect() *GoDataSelectQuery           { return o.Select }
+func (o *GoDataQuery) GetOrderBy() *GoDataOrderByQuery         { return o.OrderBy }
+func (o *GoDataQuery) GetTop() *GoDataTopQuery                 { return o.Top }
+func (o *GoDataQuery) GetSkip() *GoDataSkipQuery               { return o.Skip }
+func (o *GoDataQuery) GetCount() *GoDataCountQuery             { return o.Count }
+func (o *GoDataQuery) GetInlineCount() *GoDataInlineCountQuery { return o.InlineCount }
+func (o *GoDataQuery) GetSearch() *GoDataSearchQuery           { return o.Search }
+func (o *GoDataQuery) GetCompute() *GoDataComputeQuery         { return o.Compute }
+func (o *GoDataQuery) GetFormat() *GoDataFormatQuery           { return o.Format }
+func (o *GoDataQuery) AddExpandItem(item *ExpandItem) {
```

Comment:
Add code comment explaining the purpose. What does it mean to add an ExpandItem to a GoDataQuery? Add example

Reply:
Added